### ISSUE TITLE
Please Add on next update OnReceivedError 

### ISSUE THREAD

**Issue by Doughouzlight** [SKLn-Rad/Xam.Plugin.Webview#14]
Hi Thank you for your plugin 

I have a suggestion if you can add it on next update 
because i need it soo much in my application and i am using your Plugin at the main time

its fast not like the Xlabs hybrid view it leaks the memory  in mobile devices.

i cannot figure it out how to handle error connection during the browsing
but i have the code but cannot implement  it on your plugin 

if you can tell me how can i handel it i will be very thankful 

if i should wait tell next update i will be also thankful .

` public override void OnReceivedError(Android.Webkit.WebView view, ClientError errorCode, string description, string failingUrl)
            {
                base.OnReceivedError(view, errorCode, description, failingUrl);
                System.Console.WriteLine("\nIn AppName.Droid.FormsWebViewRenderer - Error\nError Code: {0}\nDescription: {1}\nFailing URL: {2}\n", errorCode, description, failingUrl); //TODO: Do something more useful

                //view.LoadData(/* Add your custom error message here along with the 'description' parameter or whatever */, "text/html", "UTF-8");                                         //Load custom errors
            }`

Regards 

Allen

**Comment by SKLn-Rad** [comment]
Hey Allen,

The errors reported by OnReceivedError are unique to the Android platform and therefore are quite tricky to mimic on the iOS and Windows platforms.
To get around this I have updated the code to throw a HttpStatusCode when an error is occurred during the navigation process. You should be able to catch them now in version 1.4.0 if this is okay enough for you?

Just use the OnNavigationError event.

If you still need the unique errors in OnReceivedError then you can always get the WebViewClient from the Android Control.

Do let me know if this is okay as then I can close this issue.
Regards,
Ryan

**Comment by Doughouzlight** [comment]
Hi Ryan

Thank you for your respond 

At the main time, i have a question is your function able to catch browser error like 

1- timeout 
2-404 not found 
3-502 

i am using your web view on my application it looks now like hybrid but if the user press and receives an error from the web view i should set my own style when the website navigates to 404 as an example if i can catch it by your OnNavigationError event then it will be perfect.

Regards 
Allen

**Comment by SKLn-Rad** [comment]
Yes if by timeout you mean a 408 error which should be handled natively.
Of course this is leveraging the native API's so its probably worth a check, especially as functionality for this in Android was only adding in version M so I have had to work around it using the ApacheHttp libraries 👍 

Let me know how you get on! you should be able to swap the content type and display a custom HTML string/file when you get the error now.

**Comment by Doughouzlight** [comment]
Hi Ryan 

I have added your new update to my application 
the application crashes when it opens your plugin 

I added breakpoint and i noticed it crashes on InitializeComponent() function 

this happens after the New update 

Regards
Allen

**Comment by SKLn-Rad** [comment]
Hey Allen, just wondering before I debug. What Android version are you using and what resource type are you using? HttpString, File or Internet?

I think I can potentially see any issue if you're not using the internet resource type on versions of Android lower than marshmallow but I am not too sure.

Did you get a stack trace I can see?

Regards,
Ryan

**Comment by Doughouzlight** [comment]
Hi Ryan 

I am using android 6.0

last xamarin and last xamarin forms

i am loading the browser via internet (Website)

The error says error token e34e34000

i have also tried on new project same problem during the InitializeComponent() it crashes 

Its just on your last version 

when i installed 1.3 its ok working normally 

Regards 

Allen

**Comment by SKLn-Rad** [comment]
Oh there was a change i don't think i documented for another user. If you have come from 1.3 can you make sure you're calling your JavaScript after the OnContentLoaded event instead of the OnNavigationCompleted event. This is to make sure the DOM is ready before i handle the bridge.

I will update the github to reflect this!

**Comment by Doughouzlight** [comment]
Hi Ryan 

Still waiting from you 

Any update from you ? 

Regards

**Comment by SKLn-Rad** [comment]
Hey man, I thought I responded?
Would it be possible for you to create a sample project so that I can get a better understanding and try to debug for you faster?

Regards,
Ryan

**Comment by Doughouzlight** [comment]
Hi 

I seem i had fixed the problem by uninstalling Full plugin after that installing 1.4.1 version 

if any person is updating from 1..3.3 to 1.4.1 it will break he needs to remove the plugin totally from the nugget after that he needs to install it again.

Here is the link of the sample project >> https://www.sendspace.com/file/t35bz4

I was very happy when i stop crashing problem, but after that, i noticed new bug 

The last version is not opening any page or any website its always blank Page

i have tested on version 1.3.3 its ok but the new version 1.4.1 its always blank 

Let me know what i need to do thank you for your Plugin it saved my life )) .

Regards 

Allen

**Comment by Doughouzlight** [comment]
Hi Ryan

Hope you are still alive ;)) .

Regards

**Comment by SKLn-Rad** [comment]
Hey Allen, huge apologies on the delay. A couple of my partners dropped out of some apps we were working on so I have been covering like mad recently and haven't been on open source stuff much!

I found two issues with the sample app.
1) The Internet permission was never added to the app manifest
2) There was no guarantee that the delegates will return on the UI thread, which caused the webpage to not load sometimes. I have forced this to always use the UI thread on 1.4.2 and have updated your sample app which can be found here: https://www.sendspace.com/file/ntksy5

Once again, huge apologies. I hope this works for you now.
Regards,
Ryan
